Documentation/Testing/ProjectTesting.rst

```diff
@@ -6,262 +6,234 @@
+instead of parallel execution.
 
+Project may have different places for tests
 
-Project site-introduction
+*   local path extension tests `packages/*/Tests/*`
+*   global (root) tests `Tests/*`
+
+The Core mono repository is basically a project setup, having local path
+extensions in `typo3/sysexts/*` instead of the more known and lived `packages/*`
+project folder structure.
+
+.. _testing-projects-structure:
+
+Project structure
+=================
+
+We assume a project structure similar to `tf-basics-project <https://github.com/sbuerk/tf-basics-project>`__
+here. If you are using another structure, you have to adjust some scripts.
+
+*   :ref:`TYPO3 installation with Composer <t3start:install>` and default paths
+*   :ref:`DDEV <t3start:installation-ddev-tutorial>` is used for local development
+*   Local extensions and the site package are situated in :path:`packages`
+
+..  directory-tree::
+    :caption: Initial directory tree in version control
+    :level: 1
+
+    *   :path:`.ddev`
+    *   :path:`config`
+    *   :path:`packages`
+    *   :file:`composer.json`
+    *   :file:`composer.lock`
+
+The :file:`composer.json` looks like this:
+
+..  literalinclude:: _ProjectTesting/_composer.json
+    :caption: Example project composer.json before testing
+
+.. _testing-projects-installation:
+
+Install testing dependencies
+============================
+
+As a bare minimum it is suggested to use
+
+*   One coding style fixer for PHP, for example :composer:`friendsofphp/php-cs-fixer`
+*   One static code analyzer for PHP, for example :composer:`phpstan/phpstan`
+
+Depending on the complexity of your project you might need:
+
+*   :composer:`phpunit/phpunit`, if there is any PHP code of a complexity that
+    should be tested.
+*   Testing of scss, TypeScript or JavaScript (not covered here)
+*   Linting of YAML, XML, TypoScript (not covered here)
+*   :ref:`Writing acceptance tests <t3coreapi:testing-writing-acceptance>`
+
+You can install all these tools as development dependencies. They will then not
+be installed on your production system when Composer is executed with option
+`--no-dev` during deployment:
+
+..  code-block:: bash
+    :caption: Composer installation **during deployment**
+
+    composer install --no-dev
+
+For TYPO3 project you can use the package :composer:`typo3/coding-standards`
+which bundles :composer:`friendsofphp/php-cs-fixer`, :composer:`phpstan/phpstan`,
+:composer:`phpunit/phpunit` and some other helpful tools and comes with useful
+configurations for these.
+
+..  code-block:: bash
+    :caption: Require development dependencies
+
+    composer req --dev typo3/coding-standards
+
+If you want to do Unit or Functional tests on project level you also need the
+TYPO3 testing framework:
+
+..  code-block:: bash
+    :caption: Require development dependencies
+
+    composer req --dev typo3/coding-standards typo3/testing-framework
+
+.. _testing-projects-configuration:
```

Comment:
Sure, not a must have, but first title says "Differences between project and extension setting". So, I prefer using "project" here:

```suggestion
.. _testing-project-configuration:
```